src/System.Threading.Tasks.Extensions/src/System/Runtime/CompilerServices/AsyncBuilderAttribute.cs

```diff
@@ -5,14 +5,20 @@
 namespace System.Runtime.CompilerServices
 {
     /// <summary>
-    /// Indicates the async method builder that should be used by the C# compiler to build
-    /// the attributed type when used as the return type of an async method.
+    /// Indicates the async method builder that should be used by a language compiler to
+    /// build the attributed type when used as the return type of an async method.
     /// </summary>
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Interface, Inherited = false, AllowMultiple = false)]
-    internal sealed class AsyncBuilderAttribute : Attribute
+    public sealed class AsyncBuilderAttribute : Attribute
```

Comment:
> I didn't test enums+delegates, but I'm pretty sure they can be made tasklike.

@ljw1004, do you want me to add those to the attribute targets?  I guess there's little argument not to.